src/Query.cpp

```diff
@@ -17,6 +17,7 @@ Query::~Query()
 {
     if(m_query_handle)
     {
+        m_query_handle->Reset();
```

Comment:
I think Reset() should be called in QueryPool::returnQuery() so it is consistently done every time returnQuery() is called

Reply:
Removed from Query destructor and added to QueryPool::returnQuery. I also left the call to Reset in the QueryHandle destructor.